lib/ts/fetch.ts

```diff
@@ -567,167 +595,201 @@ export async function onInvalidClaimResponse(response: ResponseWithBody) {
+    // since the active session may be using cookies. By default, we want to allow users to continue these sessions.
+    // The new session preference should be applied at the start of the next session, if the backend allows it.
 
-    if (idRefreshToken === "remove" && status === "EXISTS") {
-        // we check for wasLoggedIn cause we don't want to fire an event
-        // unnecessarily on first app load or if the user tried
-        // to query an API that returned 401 while the user was not logged in...
-        if (statusCode === AuthHttpRequest.config.sessionExpiredStatusCode) {
-            logDebugMessage("setIdRefreshToken: firing UNAUTHORISED event");
-            AuthHttpRequest.config.onHandleEvent({
-                action: "UNAUTHORISED",
-                sessionExpiredOrRevoked: true,
-                userContext: {}
-            });
+    const accessToken = await getTokenForHeaderAuth("access");
+    const refreshToken = await getTokenForHeaderAuth("refresh");
+
+    // We don't add the refresh token because that's only required by the refresh call which is done with fetch
+    // Still, we only add the Authorization header if both are present, because we are planning to add an option to expose the
+    // access token to the frontend while using cookie based auth - so that users can get the access token to use
+    if (accessToken !== undefined && refreshToken !== undefined) {
+        // the Headers class normalizes header names so we don't have to worry about casing
+        if (clonedHeaders.has("Authorization")) {
+            logDebugMessage("setAuthorizationHeaderIfRequired: Authorization header defined by the user, not adding");
         } else {
-            logDebugMessage("setIdRefreshToken: firing SIGN_OUT event");
-            AuthHttpRequest.config.onHandleEvent({
-                action: "SIGN_OUT",
-                userContext: {}
-            });
+            logDebugMessage("setAuthorizationHeaderIfRequired: added authorization header");
+            clonedHeaders.set("Authorization", `Bearer ${addRefreshToken ? refreshToken : accessToken}`);
         }
+    } else {
+        logDebugMessage("setAuthorizationHeaderIfRequired: token for header based auth not found");
     }
+}
 
-    if (idRefreshToken !== "remove" && status === "NOT_EXISTS") {
-        logDebugMessage("setIdRefreshToken: firing SESSION_CREATED event");
-        AuthHttpRequest.config.onHandleEvent({
-            action: "SESSION_CREATED",
-            userContext: {}
-        });
+async function saveTokensFromHeaders(response: Response) {
+    logDebugMessage("saveTokensFromHeaders: Saving updated tokens from the response headers");
+
+    const refreshToken = response.headers.get("st-refresh-token");
+    if (refreshToken) {
+        logDebugMessage("saveTokensFromHeaders: saving new refresh token");
+        const [value, expiry] = refreshToken.split(";");
```

Comment:
On second thought, sending the expiry from the backend for header based auth (as part of the value for st-refersh-token or st-access-token) is very strange. Please just send the access token and refresh token without the `;` + time after.

The reason is cause it affects developer experience when using postman. People will call the login API and get back headets (since that the default auth mode), and then they will have to copy the access token and remove the `;<time>` part from it and then paste it in the authorization header. This is a bad DX.

On the frontend, we should just set the access and refresh token to be 100 years for header based auth.